internal/provider/dedicated_cluster_data_source.go

```diff
@@ -0,0 +1,330 @@
+					},
+					"is_default_group": schema.BoolAttribute{
+						MarkdownDescription: "Indicates if this is the default group.",
+						Computed:            true,
+					},
+					"state": schema.StringAttribute{
+						MarkdownDescription: "The state of the node group.",
+						Computed:            true,
+					},
+					"endpoints": schema.ListNestedAttribute{
+						MarkdownDescription: "The endpoints of the node group.",
+						Computed:            true,
+						NestedObject: schema.NestedAttributeObject{
+							Attributes: map[string]schema.Attribute{
+								"host": schema.StringAttribute{
+									MarkdownDescription: "The host of the endpoint.",
+									Computed:            true,
+								},
+								"port": schema.Int32Attribute{
+									MarkdownDescription: "The port of the endpoint.",
+									Computed:            true,
+								},
+								"connection_type": schema.StringAttribute{
+									MarkdownDescription: "The connection type of the endpoint.",
+									Computed:            true,
+								},
+							},
+						},
+					},
+				},
+			},
+			"tikv_node_setting": schema.SingleNestedAttribute{
+				MarkdownDescription: "Settings for TiKV nodes.",
+				Computed:            true,
+				Attributes: map[string]schema.Attribute{
+					"node_spec_key": schema.StringAttribute{
+						MarkdownDescription: "The node specification key.",
+						Computed:            true,
+					},
+					"node_count": schema.Int32Attribute{
+						MarkdownDescription: "The number of nodes in the cluster.",
+						Computed:            true,
+					},
+					"storage_size_gi": schema.Int32Attribute{
+						MarkdownDescription: "The storage size in GiB.",
+						Computed:            true,
+					},
+					"storage_type": schema.StringAttribute{
+						MarkdownDescription: "The storage type.",
+						Computed:            true,
+					},
+					"node_spec_display_name": schema.StringAttribute{
+						MarkdownDescription: "The display name of the node spec.",
+						Computed:            true,
+					},
+				},
+			},
+			"tiflash_node_setting": schema.SingleNestedAttribute{
+				MarkdownDescription: "Settings for TiFlash nodes.",
+				Computed:            true,
+				Attributes: map[string]schema.Attribute{
+					"node_spec_key": schema.StringAttribute{
+						MarkdownDescription: "The node specification key.",
+						Computed:            true,
+					},
+					"node_count": schema.Int32Attribute{
+						MarkdownDescription: "The number of nodes in the cluster.",
+						Computed:            true,
+					},
+					"storage_size_gi": schema.Int32Attribute{
+						MarkdownDescription: "The storage size in GiB.",
+						Computed:            true,
+					},
+					"storage_type": schema.StringAttribute{
+						MarkdownDescription: "The storage type.",
+						Computed:            true,
+					},
+					"node_spec_display_name": schema.StringAttribute{
+						MarkdownDescription: "The display name of the node spec.",
+						Computed:            true,
+					},
+				},
+			},
```

Comment:
Looks to me some attributes are missing. And we only embed default node group right?